draft-nro-bulk-rdap.md

```diff
@@ -48,184 +48,294 @@ feature of this service.
 
 # Bulk Data Format {#bulk_data_format}
 
-The data returned for a Bulk RDAP request ((#bulk_rdap_url)) is a JSON object comprising JSON members for metadata and
-JSON data for a single RDAP object class.
+The data returned for a Bulk RDAP request ((#bulk_rdap_url)) is a sequence of JSON objects, each followed by a newline
+('\n') character. Such a sequence is commonly known as [JSON Lines](https://jsonlines.org/), or newline-delimited JSON,
+and makes bulk data streaming computationally efficient for the clients.
+The first object in the returned sequence provides the metadata information for the bulk data, and the following objects
+are data objects for one or more RDAP object classes ([@!RFC9083, section 5]).
 
 ## Metadata
 
-The following JSON members for metadata MUST be included:
-
-* "rdapConformance" -- An array of strings ([@!RFC9083, section 4.1]) to signal the RDAP extensions the data is based
-  on.
-* "serial" -- An unsigned 32-bit number for sequencing the data snapshots. It is RECOMMENDED to follow the serial
-  number-wrapping arithmetic from [@!RFC1982].
-
-## Data for an Object Class
-
-The JSON data for an RDAP object class is an "objects" member -- a JSON array comprising all the objects for that class.
-It is RECOMMENDED that an RIR provide bulk data for IP Network, Autonomous System Number, Domain, Nameserver, and Entity
-object classes ([@!RFC9083, section 5]).
-
-Furthermore, within the JSON array:
-
-* The "self" links for each primary object and the secondary objects it contains MUST be included to have parity with
-  the "ref" element in the Bulk Whois.
-* For compactness, it is NOT RECOMMENDED to include details for a secondary object beside its "self" link, "handle" when
-  defined, and relationship to the primary object (e.g., using the "roles" member in an Entity object to relate to an IP
-  Network object).
-* An "rdapConformance" member MAY be included at the top of a primary object if the RDAP extensions used to produce it
-  are different from those listed in the "rdapConformance" member for metadata.
-
-## Example
-
-Here is an elided example of a JSON object containing bulk data for the IP Network object class:
+The metadata JSON object MUST include the following members:
+
+* "extensionId" -- the "nroBulkRdap1" string (see (#rdap_extensions_registry))
+* "versionId" -- a version 4 Universally Unique IDentifier (UUID, [@!RFC9562, section 5.4]) string that MUST be the same
+  for all the bulk data generated for various RDAP object classes at a particular interval (say, on a particular day)
```

Comment:
It may be worth making this less abstract, so something like "that MUST be the same for bulk data that was generated by the producer at a given time, such that if multiple bulk files with the same versionId are combined, the result is consistent as among its records".

Reply:
Much better!